packages/react/src/components/ThemeProvider/index.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+import { createTheme, Theme } from '@aws-amplify/ui';
+
+import { ThemeContext, ColorMode } from './ThemeContext';
+
+interface ThemeProviderProps {
+  children: React.ReactNode;
+  colorMode?: ColorMode;
+  theme?: Theme;
+  root?: boolean;
+}
+
+export { ThemeContext, ColorMode };
+
+export function ThemeProvider({
+  children,
+  colorMode,
+  theme,
+  root,
+}: ThemeProviderProps) {
+  // createTheme will always return a fully formed theme object,
+  // extending from the default theme. Doing this allows
+  // users to still use the `useTheme` hook even if they do not
+  // provide a theme
+  const webTheme = createTheme(theme);
+  const { name, cssText } = webTheme;
+
+  React.useEffect(() => {
+    if (root) {
+      if (document && document.documentElement) {
```

Comment:
Could use optional chaining here?
```suggestion
      if (document?.documentElement) {
```

Reply:
@dbanksdesign  I withdraw my comment given: https://blog.jim-nielsen.com/2022/a-web-for-all/